tests/contrib/subjects/test_subjects_service.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Test the subject vocabulary service."""
+
+import pytest
+from invenio_pidstore.errors import PIDAlreadyExists, PIDDeletedError
+from marshmallow.exceptions import ValidationError
+from sqlalchemy.exc import IntegrityError
+
+
+def test_controlled_subject_simple_flow(
```

Comment:
I think we have a difference on use cases... you are pointing towards "users can add subjects to a subtype" and for affiliations what we did was "free text/custom ones are stored in the record". That means:
- No aggregations possible
- No sharing between records